### ISSUE TITLE
Feature Request: Shortcuts While Dragging

### ISSUE THREAD

**Issue by Ichisich** [tablacus/TablacusExplorer#734]
Maybe there is already a way I'm not aware of, but it seems shortcuts get completely disabled while holding an element on left click.
Switching Tabs while drag'n'drop via shortcut is much faster than dragging it over the tabs itself.

**Comment by JoxCoswit** [comment]
In the new version of **Tab plus 1.97** the option "**Drop to Folder**" was recently added.
_Try to make it look like the picture below._

Is that according to your request?

![Drop to Folder](https://github.com/tablacus/TablacusExplorer/assets/60333254/6557abce-0c42-40bc-9ab9-914d7114df19)

**Comment by Ichisich** [comment]
Thanks for your reply.

Yes I'm aware of those options and use them currently to work around the issue, but this approach is noticeably slower than shortcut switching.

**Comment by JoxCoswit** [comment]
Maybe someone knows a better solution.
I know that with the help of scripts, shortcuts can be made to switch tabs.

Move Tab to left
https://gist.github.com/tablacus/fea97a13a4627fc7485012e6da647015
Move Tab to right
https://gist.github.com/tablacus/831947fc23f1e6f94eddc323dcfb542c

**Comment by Ichisich** [comment]
Yea those are useful, but in context of drag'n'drop simple tab switching and mousewheel scrolling would be enough.
Ideally, an option to enable shortcuts while dragging in general would be possible.

**Comment by Ichisich** [comment]
After further testing, it's not the drag actions which disables every other input, but the mouse left click itself.
As long as mouse left click is held, no other input than ESC to cancel and ctrl, alt, shift to change the operation is accepted.
Other mouse inputs aren't treated as such.

**Comment by JoxCoswit** [comment]
What other options would you need for mouse operations in terms of dragging shortcuts?

**Comment by Ichisich** [comment]
I don't drag shortcuts, I drag files and use shortcuts, which are keyboard inputs btw, to accelerate the process.
But holding left click disables almost all keyboard inputs in Tablacus.

**Comment by JoxCoswit** [comment]
Yes. it's clearer to me what you meant and what you wanted to achieve.
Maybe it's possible even now but I'm not an expert on how to do it.

This might be possible with some kind of Multi process file operations/executions add-ons.
Which would then enable receiving parallel commands-something like that.

It seems to me that at that moment one excludes the other.

Regards, 
Jox.

**Comment by ccfs** [comment]
The easiest workaround is to use a shorter timeout before switching tab while dragging. The [patch](https://github.com/tablacus/TablacusExplorer/issues/692) version has been enhanced to let you specify a timeout (e.g. 50 ms) on the options page of "Tab plus" add-on.

**Comment by Ichisich** [comment]
@ccfs 

Can I just drag the contents of the tabplus addon folder from the patch into my tablacus installation or do I have to run a script?

**Comment by ccfs** [comment]
You must use TE 25.6.8 as the base version and run a command script to patch it. Steps:

1. Install a new copy of [TE250608](https://github.com/tablacus/TablacusExplorer/releases/download/25.6.8/te250608.zip).
2. Delete the addons folder from the new copy.
3. Copy addons and config (and icons, if any) folders from your current TE to the new copy.
4. From Command Prompt:
	i. Run `Patch.cmd` or `PatchAdd.cmd` with the path of the new copy.
	ii. If you want to turn off WebView2 extension, go to the directory of the new copy and run `SwitchWebView2.cmd OFF`.

**Comment by Ichisich** [comment]
@ccfs 

Got it, thanks.
Reducing the timeout makes tab switching by dragging more responsive indeed.

In case, that you are the creator of the patch version, could you add an option to tabplus, that tab switching gets disabled while holding SHIFT, CTRL or ALT?
<img width="820" height="93" alt="Image" src="https://github.com/user-attachments/assets/758e181b-763a-4415-97dc-d2e00029e00c" />

**Comment by ccfs** [comment]
Sorry, please clarify your request.

1. Do you want to TEMPORARILY disable tab switching if Shift / Ctrl /Alt is holding, even when "Open" is ticked?

2. How is that related to "Drop to folder" option, which you mark as "Default Behaviour"?

**Comment by Ichisich** [comment]
Case 1:
I drag an element over a tab -> List view switches to that tab.

Case 2,  while holding Shift (move element), Ctrl (copy element), Alt (link element);
I drag an element over a tab while holding a modal key -> List view keeps on current tab but the corresponding action (copy, move, link) gets applied.

**Comment by ccfs** [comment]
Which result do you expected? A or B?

<img width="1321" height="337" alt="Image" src="https://github.com/user-attachments/assets/7cc3fac3-140f-4c2f-82f4-d11392d06aa7" />

I will change "Drop to folder" check box to a 3-state option if the requested result is B.

**Comment by Ichisich** [comment]
For may use case, `open` and `drag to folder` are both enabled, `A` and `B` are interchangeable.

**Comment by ccfs** [comment]
I have enhanced the patch to do result A.
Please choose "Suspend when Shift/Ctrl/Alt key pressed" on the options page of "Tab plus" add-on to enable the function.

**Comment by Ichisich** [comment]
Works flawless. Thanks, @ccfs
